ui_test/src/comments.rs

```diff
@@ -63,6 +68,45 @@ impl Condition {
     }
 }
 
+#[derive(Debug, PartialEq, PartialOrd)]
+pub(crate) enum ErrorLines {
+    Single(ErrorLine),
+    /// If there is more than one entry, the diagnostic is expected to show up in exactly one
+    /// of the lines. This is useful for when a diagnostic is "flaky" in its position. Often
+    /// diagnostics can occur on different lines depending on the platform.
+    Multiple {
+        group: usize,
+        lines: Vec<ErrorLine>,
+    },
+}
+
+impl ErrorLines {
+    pub fn matches<'a, T>(
+        &self,
+        messages: &'a mut [Vec<T>],
+    ) -> (Option<(&'a mut Vec<T>, usize)>, usize) {
+        match self {
+            &ErrorLines::Single(ErrorLine { line, definition_line }) => {
+                let line = messages.get_mut(line).map(|msgs| (msgs, line));
+                (line, definition_line)
+            }
+            ErrorLines::Multiple { lines, .. } => {
+                // Cannot use `find` here or even `if let Some` within the loop, because the borrow checker
+                // thinks we borrow `messages` mutably twice in those cases.
+                for &ErrorLine { line, .. } in lines {
+                    if line < messages.len() {
+                        return (
+                            messages.get_mut(line).map(|msgs| (msgs, line)),
+                            lines[0].definition_line,
+                        );
+                    }
+                }
+                (None, lines[0].definition_line)
```

Comment:
I have no idea why this fall-back to `lines[0]` makes any sense.

Reply:
It's not a fall back, it's more of a failure case... but since it's all going away anyway...